rules/azurerm_linux_virtual_machine_invalid_admin_username.go

```diff
@@ -0,0 +1,70 @@
+package rules
+
+// Please see: https://docs.microsoft.com/en-us/azure/virtual-machines/linux/faq#what-are-the-username-requirements-when-creating-a-vm
+
+import (
+	"fmt"
+
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-azurerm/project"
+)
+
+// AzurermLinuxVirtualMachineInvalidAdminUserNameRule ensures that invalid usernames are not set.
+type AzurermLinuxVirtualMachineInvalidAdminUserNameRule struct {
+	resourceType  string
+	attributeName string
+}
+
+// NewAzurermLinuxVirtualMachineInvalidAdminUserNameRule returns new rule with default attributes
+func NewAzurermLinuxVirtualMachineInvalidAdminUserNameRule() *AzurermLinuxVirtualMachineInvalidAdminUserNameRule {
+	return &AzurermLinuxVirtualMachineInvalidAdminUserNameRule{
+		resourceType:  "azurerm_linux_virtual_machine",
+		attributeName: "admin_username",
+	}
+}
+
+// Name returns the rule name
+func (r *AzurermLinuxVirtualMachineInvalidAdminUserNameRule) Name() string {
+	return "azurerm_linux_virtual_machine_invalid_admin_username"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *AzurermLinuxVirtualMachineInvalidAdminUserNameRule) Enabled() bool {
+	return true
+}
+
+// Severity returns the rule severity
+func (r *AzurermLinuxVirtualMachineInvalidAdminUserNameRule) Severity() string {
+	return tflint.ERROR
+}
+
+// Link returns the rule reference link
+func (r *AzurermLinuxVirtualMachineInvalidAdminUserNameRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check checks the pattern is valid
+func (r *AzurermLinuxVirtualMachineInvalidAdminUserNameRule) Check(runner tflint.Runner) error {
+	return runner.WalkResourceAttributes(r.resourceType, r.attributeName, func(attribute *hcl.Attribute) error {
+		var val string
+		err := runner.EvaluateExpr(attribute.Expr, &val)
+
+		return runner.EnsureNoError(err, func() error {
+			valid, err := isValidVMAdminUserName(val)
+			if err != nil {
+				panic(err)
```

Comment:
it seems to cause a panic when `admin_username = ""`. If a panic occurs, the process will crash, so I think it's better to just assume that an invalid name was used and emit the issue.